### PR TITLE
added support for svg stroke width change and viewBox dimension

### PR DESCRIPTION
**Pull request by NeedNap**
I modified your library in order to implement the following functionalities:
- added the possibility to get the svg viewBox dimensions reading them from the relative tag
- added the strokeWidth property to allow the svg stroke change programmatically
 
I would like to find these new features in the next release of the library.
Thank you for your great job.

### CODE REVIEW

**Comment by arielelkin** [comment]
Hi @NeedNap, thank you for your PR! 

Could you please make sure that this passes our CI? (it currently indicates checks have passed but that's not the case if you look at the [CI logs](https://github.com/pocketsvg/PocketSVG/pull/181/checks?check_run_id=1188744882) ). 

In the modified SVGBezierPath.h methods, could you please make the viewBox parameter optional?

**Comment by arielelkin** [comment]
More generally, PRs adding support for SVG attributes shouldn't introduce breaking changes (i.e. people upgrading from the current version to this new version should be able to build without issues – currently they need to modify all their calls to `CGPathsFromSVGString`, `pathsFromSVGString`, etc.)

**Comment by NeedNap** [comment]
You are right: I made these changes quickly for my personal project without thinking about backward compatibility

**Comment by NeedNap** [comment]
I fixed it by remove the "unused variable 'strViewBox'" warning

File: Sources/SVGBezierPath.mm [hunk 1]
```diff
@@ -68,6 +103,20 @@ + (void)resetCache
     return paths;
 }
 
++ (NSArray<SVGBezierPath*> *)pathsFromSVGString:(NSString * const)svgString viewBox:(CGRect *)viewBox
```
Comment (arielelkin) [inline Sources/SVGBezierPath.mm:106]: why a _pointer_ to a CGRect? This creates the need to pass in an `UnsafeMutablePointer<CGRect>` in Swift.

**Review by arielelkin — COMMENTED** [review]
Hey @NeedNap, please let me know your thoughts on my comments 😀

File: Sources/SVGBezierPath.mm (hunk 1 above)
Comment (NeedNap) [reply Sources/SVGBezierPath.mm:106]: I used this pointer to get the viewBox info and put this value in the _svg_viewBoxCache variable. I don't know if this part can be write in a better way :)

File: Sources/include/SVGBezierPath.h [hunk 2]
```diff
@@ -51,11 +50,22 @@ FOUNDATION_EXTERN void SVGDrawPathsWithBlock(NSArray<SVGBezierPath*> * const pat
...
 
 
+/*!
+ * @brief Returns an array of SVGBezierPaths given an SVG's URL.
+ *
+ * @param aURL The URL from which to load an SVG.
+ * @param viewBox The SVG viewBox tag rect.
+ *
+ */
++ (NSArray<SVGBezierPath*> *)pathsFromSVGAtURL:(NSURL *)aURL viewBox:(CGRect *)viewBox;
```
[the remark this answers is not in the document]
Comment (NeedNap) [reply Sources/include/SVGBezierPath.h:60]: I don't understand this question. Do you mean if the SVGBezierPath class? This code read the viewBox tag from the SVG file and return to the caller (by the viewBox pointer) these values in a CGRect struct.

[the remark this answers is not in the document]
Comment (arielelkin) [reply Sources/include/SVGBezierPath.h:60]: Ah ok now I understand the intention of your code and why you pass a pointer to a CGRect (I initially thought you wanted the caller to pass a CGRect as the viewBox to render the SVG in). 

If the caller wants to read the `viewBox` tag in the SVG file, what if we expose this property on `SVGBezierPath`:
```
@property(nonatomic, readonly) CGrect *viewBox;
```

This way, no need to pass a pointer.

[the remark this answers is not in the document]
Comment (NeedNap) [reply Sources/include/SVGBezierPath.h:60]: Ok, you are right, but I use the SVGBezierPath static function to get the viewBox information so I can't store the value in a class property.
Do you have any suggestions?

[the remark this answers is not in the document]
Comment (arielelkin) [reply Sources/include/SVGBezierPath.h:60]: Ciao Alessio, sorry for the delay, I'll be able to take a look at this soon.

**Comment by arielelkin** [comment]
Alessio, sorry for the long wait. 

I've been thinking about exposing viewBox as part of the PocketSVG API. The `viewBox` attribute defines the scaling and positioning of a graphic within its viewport. So I would expect an SVG renderer like PocketSVG to render an SVG accounting for its `viewBox` attribute. It shouldn't be the user's responsibility to check what an SVGImageView's or SVGLayer's `viewBox` is and then manually transform the graphic.

Did you have a similar use case or something else in mind?

In any case, if we're not applying the `viewBox` attribute to the render we should go for a simpler API. What do you think about adding this method to SVGBezierPath? 

```objc
+ (CGRect)viewBoxFromSVGAtURL:(NSURL *)aURL;
```

This would do a simple XML parsing of the SVG string and convert the viewBox attribute into a CGRect (no need for caching it)

Let me know what you think!

**Comment by NeedNap** [comment]
@arielelkin I think what you propose is the best solution. The only con I see is that the svg file will be parsed two times.
In my use case I need this information to properly size the svg container, otherwise I can't know the exact svg dimensions.

Let me know if you need more explanations.

**Comment by arielelkin** [comment]
@NeedNap I found a simple way to parse the SVG just once and save the `viewBox` attribute, exposing it in SVGLayer and SVGImageView. It'll also show up as any individual SVGBezierPath's `viewBox` attribute. 

Please review if this suits your need:
https://github.com/pocketsvg/PocketSVG/pull/185

Could you also please modify this PR (#181) so that it only adds support for svg stroke width change?

**Comment by arielelkin** [comment]
@NeedNap I hope you're well! 

Are you ok to modify this PR so that it only adds support svg stroke width change?

**Pull request closed without merging**